spec/components/custom_wizard/mapper_spec.rb

```diff
@@ -250,4 +274,75 @@
       user: user1
     ).perform).to eq(false)
   end
+
+  context "Output Templating" do
+    it "passes the correct values to the template" do
+      template = "w{step_1_field_1}"
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"])
+    end
+
+    it "treats replaced values as string literals" do
+      template = '{{ "w{step_1_field_1}" | size }}'
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "allows the wizard values to be used inside conditionals" do
+      template = <<-LIQUID
+        {%- if "w{step_1_field_1}" contains "ello" -%}
+          Correct
+        {%- else -%}
+          Incorrect
+        {%-endif-%}
+      LIQUID
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq("Correct")
+    end
+
+    it "can access data passed to render method as variable" do
+      template = "{{step_1_field_1.size}}"
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "custom filter: 'first_non_empty' gives first non empty element from list" do
+      template = <<-LIQUID.strip
+        {%- assign entry = "" | first_non_empty: step_1_field_1, step_1_field_2, step_1_field_3 -%}
+        {{ entry }}
+      LIQUID
+      result = template_mapper_non_empty.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params_non_empty["step_1_field_3"])
+    end
+  end
```

Comment:
@fzngagan We're missing a spec that explicitly tests the case we discussed

```
{% if "step_1_field_{1|2|3|4|5}" %}
  {%= "step_1_field_{1|2|3|4|5}" %}
{% end %}
```

Reply:
To simplify this, I've create a liquid filter `first_non_empty` :)